libs/pageserver_api/src/models.rs

```diff
@@ -369,6 +369,14 @@ pub struct TenantInfo {
     pub attachment_status: TenantAttachmentStatus,
 }
 
+#[derive(Serialize, Deserialize, Clone)]
+pub struct TenantDetails {
+    #[serde(flatten)]
```

Comment:
Is this necessary for backward compatibility?  (not sure if control plane uses this endpiont)

structs like this are easier to maintain without `flatten` -- when using flatten, we have to remember that `TenantDetails` and `TenantInfo` can't have any attributes with the same name.

Reply:
Yeah, it's for backward-compatibility; I don't know if we actually need the backward-compatibility though.

I don't want to delay this PR trying to figure out whether backward-compatibility is needed or not. Do you insist?

Reply:
No, but if you could check for usage in the control plane and simplify it in a followup if it's not, that would be appreciated.